adagrams/game.py

```diff
@@ -1,13 +1,98 @@
 from random import randint
 
+LETTER_POOL = {
+    'A': 9, 'B': 2, 'C': 2, 'D': 4, 'E': 12, 'F': 2, 'G': 3, 'H': 2, 'I': 9, 
+    'J': 1, 'K': 1, 'L': 4, 'M': 2, 'N': 6, 'O': 8, 'P': 2, 'Q': 1, 'R': 6, 
+    'S': 4, 'T': 6, 'U': 4, 'V': 2, 'W': 2, 'X': 1, 'Y': 2, 'Z': 1
+}
+# Helper function 
+def build_sack_of_letters():
+    letters=[]
+    for letter, count in LETTER_POOL.items():
+        letters.extend([letter] * count)
+    
+    return letters
+
 def draw_letters():
-    pass
+    sack_of_letters =  build_sack_of_letters()
+    drawn_letters = []
+    letter_counts = {}
+    
+    while len(drawn_letters) < 10:
+        index = randint(0, len(sack_of_letters) - 1)
+        letter = sack_of_letters[index]
+        letter_counts[letter] = letter_counts.get(letter, 0) + 1
+
+        # Ensure we don't exceed the allowed count of any letter
+        if letter_counts[letter] <= LETTER_POOL[letter]:
+            drawn_letters.append(letter)
+
+    return drawn_letters
+
+# Helper function
+def build_count_letters(letter_list):
+    letter_counter = {}
+    for letter in letter_list:
+        letter_counter[letter] = letter_counter.get(letter, 0) + 1
+    return letter_counter
 
 def uses_available_letters(word, letter_bank):
-    pass
+    letter_counts = build_count_letters(letter_bank)
+
+    for letter in word.upper():
+        if letter not in letter_counts or letter_counts[letter] == 0:
+            return False
+        letter_counts[letter] -= 1
+    return True
+
 
 def score_word(word):
-    pass
+    letter_scores = {
+    "A": 1, "E": 1, "I": 1, "O": 1, "U": 1, "L": 1, "N": 1, "R": 1, "S": 1, "T": 1,
+    "D": 2, "G": 2,
+    "B": 3, "C": 3, "M": 3, "P": 3,
+    "F": 4, "H": 4, "V": 4, "W": 4, "Y": 4,
+    "K": 5,
+    "J": 8, "X": 8,
+    "Q": 10, "Z": 10
+}
+    word = word.upper()
+
+    if word == "":
```

Comment:
There are several different ways to check and see if a string is empty. While any of them are valid, one of the simplest ones is to simply check for the truthiness of a string! An empty string will always just return False.

```suggestion
    if not word:
```